internal/resources/cloudprovider/resource_aws_account_test.go

```diff
@@ -1,75 +1,116 @@
-				return nil
-			}
-		}(),
 		Steps: []resource.TestStep{
 			{
-				Config: awsAccountResourceData(),
+				Config: awsAccountResourceData(testCfg.stackID, account),
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr("grafana_cloud_provider_aws_account.test", "stack_id", testAWSAccountData.StackID),
-					resource.TestCheckResourceAttr("grafana_cloud_provider_aws_account.test", "role_arn", testAWSAccountData.RoleARN),
-					resource.TestCheckResourceAttr("grafana_cloud_provider_aws_account.test", "regions.#", strconv.Itoa(len(testAWSAccountData.Regions))),
-					resource.TestCheckResourceAttr("grafana_cloud_provider_aws_account.test", "regions.0", testAWSAccountData.Regions[0]),
-					resource.TestCheckResourceAttr("grafana_cloud_provider_aws_account.test", "regions.1", testAWSAccountData.Regions[1]),
-					resource.TestCheckResourceAttr("grafana_cloud_provider_aws_account.test", "regions.2", testAWSAccountData.Regions[2]),
+					checkAWSAccountResourceExists("grafana_cloud_provider_aws_account.test", testCfg.stackID, &gotAccount),
+					resource.TestCheckResourceAttr("grafana_cloud_provider_aws_account.test", "stack_id", testCfg.stackID),
+					resource.TestCheckResourceAttr("grafana_cloud_provider_aws_account.test", "role_arn", account.RoleARN),
+					resource.TestCheckResourceAttr("grafana_cloud_provider_aws_account.test", "regions.#", strconv.Itoa(len(account.Regions))),
+					resource.TestCheckResourceAttr("grafana_cloud_provider_aws_account.test", "regions.0", account.Regions[0]),
+					resource.TestCheckResourceAttr("grafana_cloud_provider_aws_account.test", "regions.1", account.Regions[1]),
+					resource.TestCheckResourceAttr("grafana_cloud_provider_aws_account.test", "regions.2", account.Regions[2]),
 				),
 			},
+			{
+				ResourceName:      "grafana_cloud_provider_aws_account.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
 		},
+		CheckDestroy: checkAWSAccountResourceDestroy(testCfg.stackID, &gotAccount),
 	})
 }
 
-func awsAccountResourceData() string {
+func checkAWSAccountResourceExists(rn string, stackID string, account *cloudproviderapi.AWSAccount) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[rn]
+		if !ok {
+			return fmt.Errorf("resource not found: %s\n %#v", rn, s.RootModule().Resources)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("resource id not set")
+		}
+
+		parts := strings.SplitN(rs.Primary.ID, ":", 2)
+		if len(parts) != 2 || parts[0] == "" || parts[1] == "" {
+			return fmt.Errorf("Invalid ID: %s", rs.Primary.ID)
+		}
+		accountID := parts[1]
+
+		if accountID == "" {
+			return fmt.Errorf("account id not set")
+		}
+
+		client := testutils.Provider.Meta().(*common.Client).CloudProviderAPI
+		gotAccount, err := client.GetAWSAccount(context.Background(), stackID, accountID)
+		if err != nil {
+			return fmt.Errorf("error getting account: %s", err)
+		}
+
+		*account = gotAccount
```

Comment:
why in here overwrite the ptr value with the getted account?

Reply:
this is a pattern that I copied from other packages in this repo, e.g. https://github.com/grafana/terraform-provider-grafana/blob/ad25130f6220d560d0247c98b9767b8e6960ad08/internal/resources/cloud/resource_cloud_stack_test.go#L221

the idea is that we want to check the API for the resource to no longer exist in order to ensure that it is destroyed via https://github.com/grafana/terraform-provider-grafana/blob/439626c56f114eca1f8b216d980e18b15c20130d/internal/resources/cloudprovider/resource_aws_account_test.go#L49 in order to do that, we need the ID of the resource. 

The problem is that the way the tests are set up is declarative: the Terraform config comes from https://github.com/grafana/terraform-provider-grafana/blob/439626c56f114eca1f8b216d980e18b15c20130d/internal/resources/cloudprovider/resource_aws_account_test.go#L32 and then a number of checks are run against that resource config after applied before it is destroyed at the end. Thus one way that is used throughout this repo to pass the resource ID into the `CheckDestroy` function is to have one of the test steps write the API object to a variable outside of the scope of the `resource.ParallelTest` instance https://github.com/grafana/terraform-provider-grafana/blob/439626c56f114eca1f8b216d980e18b15c20130d/internal/resources/cloudprovider/resource_aws_account_test.go#L26 so that it can be referenced by the step executed later (CheckDestroy) so that function can use the resource ID

Reply:
sounds good!